rlp/src/main/scala/io/iohk/ethereum/rlp/package.scala

```diff
@@ -44,6 +44,23 @@ package object rlp {
     override def toString: String = s"RLPValue(${Hex.toHexString(bytes)})"
   }
 
+  /** Modelise a RLPEncodable that should be binary prefixed by a raw byte.
+    *
+    * When converting this RLPEncodable to byte, the resulting value will be:
+    * prefix || prefixedRLPEncodable.toByte
+    * where || is the binary concatenation symbol.
+    *
+    * To be able to read back the data, use TypedTransaction.TypedTransactionsRLPAggregator
+    *
+    * This is for example used for typed transaction and typed receipt.
+    *
+    * @param prefix the raw byte
+    * @param prefixedRLPEncodeable the RLPEncodable to prefix with
+    */
+  case class PrefixedRLPEncodable(prefix: Byte, prefixedRLPEncodeable: RLPEncodeable) extends RLPEncodeable {
+    require(prefix >= 0, "prefix should be in the range [0; 0x7f]")
```

Comment:
isn't this up to 0xFF?

Reply:
no, 0x7f is the limit for two reasons:
- EIP-2718: TransactionType is a positive unsigned 8-bit number between 0 and 0x7f that represents the type of the transcation
- rlp encoding spec: for a single byte whose value is in the [0x00, 0x7f] range, that byte is its own RLP encoding

That's what allows us to consider a single RLPValue [0; 0x7f] as a single raw byte transaction type (and vice-versa) 

Since Byte is a signed type, the range [0; 0x7f] check become a simple positive condition.